include/aspect/postprocess/visualization/boundary_velocity_residual.h

```diff
@@ -0,0 +1,144 @@
+       */
+      template <int dim>
+      class BoundaryVelocityResidual
+        : public DataPostprocessorVector<dim>,
+		  public SimulatorAccess<dim>,
+          public Interface<dim>
+      {
+        public:
+    	  BoundaryVelocityResidual ();
+
+          void initialize () override;
+
+          void
+          evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                                std::vector<Vector<double> > &computed_quantities) const override;
+
+          /**
+           * Declare the parameters this class takes through input files.
+           */
+          static
+		  void
+		  declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           */
+          void
+		  parse_parameters (ParameterHandler &prm) override;
+
+        private:
+          bool use_spherical_unit_vectors;
+
+          bool use_gps_data;
+
+          /**
+		   * Pointer to the gplates boundary velocity model
+		   */
+          std::unique_ptr<BoundaryVelocity::internal::GPlatesLookup <dim> >  gplates_lookup;
+
+          /**
+		   * Pointer to the  ascii boundary velocity model
+		   */
+          std::unique_ptr<Utilities::AsciiDataLookup<dim>> ascii_data_lookup;
+
+          /**
+           * Directory in which the data files are present.
+           */
+          std::string ascii_data_directory;
+
+          /**
+           * Filename of data file. The file names can contain the specifiers %s
+           * and/or %c (in this order), meaning the name of the boundary and the
+           * number of the data file time step.
+           */
+          std::string ascii_data_file_name;
+
+          /**
+           * Scale the data by a scalar factor. Can be used to transform the
+           * unit of the data (if they are not specified in SI units (m/s or
+           * m/yr depending on the "Use years in output instead of seconds"
+           * parameter).
+           */
+          double ascii_scale_factor;
+
+          /**
+           * Directory in which the data files are present.
+           */
+          std::string gplates_data_directory;
+
+          /**
+           * Filename of data file. The file names can contain the specifiers %s
+           * and/or %c (in this order), meaning the name of the boundary and the
+           * number of the data file time step.
+           */
+          std::string gplates_data_file_name;
+
+          /**
+           * Scale the data by a scalar factor. Can be used to transform the
+           * unit of the data (if they are not specified in SI units (m/s or
+           * m/yr depending on the "Use years in output instead of seconds"
+           * parameter).
+           */
+          double gplates_scale_factor;
+
+          /**
+           * Parsed user input of point1 and point2 in gplates
+           */
+          Tensor<1,2> pointone;
+          Tensor<1,2> pointtwo;
+
```

Comment:
These two variables are only used inside the `initalize` function and not read in from file. As long as we do not use them anywhere else, please put them inside that function.